src/cljs/main/broadfcui/page/workspace/notebooks/notebooks.cljs

```diff
@@ -324,20 +324,32 @@
                                                                         :data-test-id "create-modal-button"
                                                                         :onClick #(swap! state assoc :show-create-dialog? true)}]]
                            :clusters clusters
-                           :reload-after-delete #(this :-get-clusters-list))]))]))
+                           :reload-after-delete #(this :-get-clusters-list-if-whitelisted))]))]))
    :component-did-mount
    (fn [{:keys [this]}]
-     (this :-get-clusters-list))
-   :-get-clusters-list
-   (fn [{:keys [props state this]}]
+     (this :-is-leo-whitelisted))
+   :-is-leo-whitelisted
+   (fn [{:keys [state this]}]
      (endpoints/call-ajax-leo
-      {:endpoint (endpoints/get-clusters-list)
-       :headers utils/content-type=json
-       :on-done (fn [{:keys [success? get-parsed-response]}]
-                  (if success?
-                    (when-not (= (:clusters @state) (get-parsed-response))
-                      (swap! state assoc :server-response {:clusters (filter #(= (get-in props [:workspace-id :namespace]) (:googleProject %)) (get-parsed-response))}))
-                    (swap! state assoc :server-response {:server-error (get-parsed-response false)}))
-                  (let [statuses (set (map #(:status %) (get-parsed-response)))]
-                    (when (or (contains? statuses "Creating") (contains? statuses "Updating") (contains? statuses "Deleting"))
-                      (js/setTimeout (fn [] (this :-get-clusters-list)) 10000))))}))})
+       {:endpoint (endpoints/is-leo-whitelisted)
+        :headers utils/content-type=json
+        :on-done (fn [{:keys [success? get-parsed-response]}]
+                   (if success?
+                     (do (swap! state assoc :is-leo-whitelisted? true)
+                         (this :-get-clusters-list-if-whitelisted))
+                     (do (swap! state assoc :is-leo-whitelisted? false)
+                       (swap! state assoc :server-response {:server-error (get-parsed-response false)}))))}))
```

Comment:
You don't need to `assoc :is-leo-whitelisted? false`, since missing keys evaluate to `nil`, which is false-y.

If you really want it, do these in one `swap`:
```cljs
(swap! state assoc :is-leo-whitelisted? false :server-response {:server-error (get-parsed-response false)})
```